src/test/jtx/Env_test.cpp

```diff
@@ -900,6 +902,92 @@ class Env_test : public beast::unit_test::suite
         pass();
     }
 
+    void
+    testSyncSubmit()
+    {
+        using namespace jtx;
+        Env env(*this);
+
+        auto const alice = Account{"alice"};
+        auto const n = XRP(10000);
+        env.fund(n, alice);
+        BEAST_EXPECT(env.balance(alice) == n);
+
+        // submit only
+        auto applyBlobTxn = [&env](RPC::SubmitSync sync, auto&&... txnArgs) {
+            auto jt = env.jt(txnArgs...);
+            Serializer s;
+            jt.stx->add(s);
+
+            Json::Value args{Json::objectValue};
+
+            args[jss::tx_blob] = strHex(s.slice());
+            args[jss::fail_hard] = true;
+            args[jss::sync] = RPC::toCString(sync);
+
+            return env.rpc("json", "submit", args.toStyledString());
+        };
+
+        auto jr = applyBlobTxn(RPC::SubmitSync::sync, noop(alice));
+        BEAST_EXPECT(jr[jss::result][jss::engine_result] == "tesSUCCESS");
+
+        jr = applyBlobTxn(RPC::SubmitSync::async, noop(alice));
+        BEAST_EXPECT(jr[jss::result][jss::engine_result] == "terSUBMITTED");
+        // Make sure it gets processed before submitting and waiting.
+        env.app().getOPs().transactionBatch(true);
+
+        auto applier = [&env]() {
+            while (!env.app().getOPs().transactionBatch(false))
+                std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        };
+        auto t = std::thread(applier);
+
+        jr = applyBlobTxn(RPC::SubmitSync::wait, noop(alice));
+        BEAST_EXPECT(jr[jss::result][jss::engine_result] == "tesSUCCESS");
+        t.join();
+
+        // sign and submit
+        auto applyJsonTxn = [&env](
+                                RPC::SubmitSync sync,
+                                std::string const secret,
+                                Json::Value&& val) {
+            Json::Value args{Json::objectValue};
+            args[jss::secret] = secret;
+            args[jss::tx_json] = val;
+            args[jss::fail_hard] = true;
+            args[jss::sync] = RPC::toCString(sync);
+
+            return env.rpc("json", "submit", args.toStyledString());
+        };
+
+        Json::Value payment;
+        auto secret = toBase58(generateSeed("alice"));
+        payment = noop("alice");
+        payment[sfSequence.fieldName] = env.seq("alice");
+        payment[sfSetFlag.fieldName] = 0;
+        auto req = payment;
+        jr = applyJsonTxn(RPC::SubmitSync::sync, secret, std::move(req));
```

Comment:
You're making a copy (`req = payment`) just to move the copy (`std::move(req)`). Why not just pass the original? Maybe pass as a `const&` instead of an r-value.

Reply:
fixed